### PR TITLE
`ROLE_ADMIN` can access site in maintenance mode

### LINKED ISSUE #1883: Maintenance Mode is active even if the user is logged in
If I active maintenance mode (maintenance_mode: true in config.yaml) this maintenance page is visible for everyone (logged in or not) instead of being just visible to users that are not logged in.

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.0.0 RC 42
| Install type                   | Composer install
| PHP version                 | 7.4
| Web server                  | Apache
| For UX/UI issues        | Safari Version 13.1.3 (15609.4.1)

### PR DESCRIPTION
**Pull request by I-Valchev**
Fixes #1883 in the master branch, due to the new permissions feature.

### CODE REVIEW

**Review by bobdenotter — CHANGES_REQUESTED** [review]
If we merge this in as-is, we're definitely going to have confused users.. An admin user might have the site on maintenance mode, they forget about it (because *they* can see the site), only to find out that other people can't. 

We should make it _visible_ that the site is in maintenance.. Perhaps add another `snippet` widget that 'll output a warning?

**Comment by bobdenotter** [comment]
Like this? ;-) 

fhBnpx2DZT

**Comment by I-Valchev** [comment]
@bobdenotter I DOOOO. Let's add it!!

Alt Text

**Comment by bobdenotter** [comment]
Here's some HTML and CSS:

```
<style>
  #bolt-maintenance-notice {
    border: 1px solid #F00;
    position: absolute;
    background-color: white;
    color: #F00;
    top: 100px;
    left: -100px;
    width: 450px;
    transform: rotateY(0deg) rotate(-45deg);
    box-shadow: 2px 0 15px rgba(0, 0, 0, 0.5);
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    font-size: 16px;
    font-weight: bold;
    line-height: 24px;
    padding: 8px;
  }
</style>

<div id='bolt-maintenance-notice'>
  <marquee>🚨 Maintenance mode is activated 🚨 Maintenance mode is activated 🚨 Maintenance mode is activated 🚨 Maintenance mode is activated 🚨 </marquee>
</div>
```

**Comment by I-Valchev** [comment]
image

File: translations/messages.en.xlf [hunk 1]
```diff
@@ -2373,7 +2373,7 @@
         <target>Please enter your password</target>
       </segment>
     </unit>
-    <unit id="o2f3RiO" name="form.empty_email">
+    <unit id="f8zXapF" name="form.empty_email">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2376]:
```suggestion
    <unit id="o2f3RiO" name="form.empty_email">
```

File: translations/messages.en.xlf [hunk 2]
```diff
@@ -2511,67 +2511,67 @@
         <target>Reset Password</target>
       </segment>
     </unit>
-    <unit id="3prMgfK" name="reset_password.email_title">
+    <unit id="dp42qbF" name="reset_password.email_title">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2514]:
```suggestion
    <unit id="3prMgfK" name="reset_password.email_title">
```

File: translations/messages.en.xlf [hunk 3]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.email_title</source>
         <target>Hi!</target>
       </segment>
     </unit>
-    <unit id="oaRkfvE" name="reset_password.email_description">
+    <unit id="xS2vvXw" name="reset_password.email_description">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2520]:
```suggestion
    <unit id="oaRkfvE" name="reset_password.email_description">
```

File: translations/messages.en.xlf [hunk 4]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.email_description</source>
         <target>To reset your password, please visit the following link</target>
       </segment>
     </unit>
-    <unit id="Afb4dal" name="reset_password.email_expire">
+    <unit id="apkSD11" name="reset_password.email_expire">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2526]:
```suggestion
    <unit id="Afb4dal" name="reset_password.email_expire">
```

File: translations/messages.en.xlf [hunk 5]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.email_expire</source>
         <target>This link will expire in %hours% hour(s).</target>
       </segment>
     </unit>
-    <unit id="9ga3FlV" name="reset_password.email_thanks">
+    <unit id="_gg3hPE" name="reset_password.email_thanks">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2532]:
```suggestion
    <unit id="9ga3FlV" name="reset_password.email_thanks">
```

File: translations/messages.en.xlf [hunk 6]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.email_thanks</source>
         <target>Cheers!</target>
       </segment>
     </unit>
-    <unit id="7d24Grt" name="reset_password.enter_pwd">
+    <unit id="harDiJR" name="reset_password.enter_pwd">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2538]:
```suggestion
    <unit id="7d24Grt" name="reset_password.enter_pwd">
```

File: translations/messages.en.xlf [hunk 7]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.enter_pwd</source>
         <target>Please enter a password</target>
       </segment>
     </unit>
-    <unit id="a2f1pra" name="label.repeat_password">
+    <unit id="rU9cSfq" name="label.repeat_password">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2544]:
```suggestion
    <unit id="a2f1pra" name="label.repeat_password">
```

File: translations/messages.en.xlf [hunk 8]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>label.repeat_password</source>
         <target>Repeat Password</target>
       </segment>
     </unit>
-    <unit id="2f19Gat" name="reset_password.not_matching_pwds">
+    <unit id="Tg5zdeo" name="reset_password.not_matching_pwds">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2550]:
```suggestion
    <unit id="2f19Gat" name="reset_password.not_matching_pwds">
```

File: translations/messages.en.xlf [hunk 9]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.not_matching_pwds</source>
         <target>The password fields must match.</target>
       </segment>
     </unit>
-    <unit id="2ahb7da" name="reset_password.minimum_length">
+    <unit id="1JZeWxG" name="reset_password.minimum_length">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2556]:
```suggestion
    <unit id="2ahb7da" name="reset_password.minimum_length">
```

File: translations/messages.en.xlf [hunk 10]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.minimum_length</source>
         <target>Your password should be at least %s characters</target>
       </segment>
     </unit>
-    <unit id="a3frsl4" name="reset_password.no_token">
+    <unit id="GUQYLn2" name="reset_password.no_token">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2562]:
```suggestion
    <unit id="a3frsl4" name="reset_password.no_token">
```

File: translations/messages.en.xlf [hunk 11]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.no_token</source>
         <target>No reset password token found in the URL or in the session.</target>
       </segment>
     </unit>
-    <unit id="hsg341k" name="reset_password.reset_successful">
+    <unit id="SjuOnue" name="reset_password.reset_successful">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2568]:
```suggestion
    <unit id="hsg341k" name="reset_password.reset_successful">
```

File: translations/messages.en.xlf [hunk 12]
```diff
@@ -2511,67 +2511,67 @@
...
       <segment>
         <source>reset_password.reset_successful</source>
         <target>Your password has been reset successfully.</target>
       </segment>
     </unit>
-    <unit id="7f2x41k" name="reset_password.problem_with_request">
+    <unit id="5Xr92sQ" name="reset_password.problem_with_request">
```
Comment (bobdenotter) [inline translations/messages.en.xlf:2574]:
```suggestion
    <unit id="7f2x41k" name="reset_password.problem_with_request">
```

**Review by bobdenotter — CHANGES_REQUESTED** [review]
We shouldn't be modifying these `id`s in the translation files.

**Review by bobdenotter — APPROVED** [review]

**Pull request merged**